pyfstat/helper_functions.py

```diff
@@ -429,6 +429,74 @@ def convert_array_to_gsl_matrix(array):
     return gsl_matrix
 
 
+def get_sft_as_numpy_array(sftfilepattern, fMin=None, fMax=None, constraints=None):
+    """
+
+    Parameters
+    ----------
+    sftfilepattern: str
+            Pattern to match SFTs using wildcards (`*?`) and ranges [0-9];
+            multiple patterns can be given separated by colons.
+    fMin, fMax: float or None
+        Restrict frequency range to `[fMin, fMax]`.
+        If None, retreive the full frequency range.
+
+    Returns
+    ----------
+    times: np.ndarray
+        The SFT start times as a dictionary of 1D arrays, one for each detector.
+    freqs: np.ndarray
+        The frequency bins in each SFT.
+        These will be the same for each SFT,
+        so only a single 1D array is returned.
+    data: np.ndarray
+        A dictionary of 2D arrays of the complex Fourier amplitudes of the SFT data
+        for each detector in each frequency bin at each timestamp.
+    """
+
+    constraints = constraints or lalpulsar.SFTConstraints()
+    if fMin is None and fMax is None:
+        fMin = fMax = -1
+    elif fMin is None or fMax is None:
+        raise ValueError("Need either none or both of fMin, fMax.")
+
+    sft_catalog = lalpulsar.SFTdataFind(sftfilepattern, constraints)
+    ifo_labels = lalpulsar.ListIFOsInCatalog(sft_catalog)
+
+    logging.info(
+        f"Loading {sft_catalog.length} SFTs from {', '.join(ifo_labels.data)}..."
+    )
+    multi_sfts = lalpulsar.LoadMultiSFTs(sft_catalog, fMin, fMax)
+    logging.info("done!")
+
+    frequencies = []
+    times = {}
+    amplitudes = {}
+
+    for ind, ifo in enumerate(ifo_labels.data):
+
+        sfts = multi_sfts.data[ind]
+
+        times[ifo] = np.array([sft.epoch.gpsSeconds for sft in sfts.data])
+        amplitudes[ifo] = np.array([np.abs(sft.data.data) for sft in sfts.data]).T
```

Comment:
I thought you wanted to _not_ take the abs?

Reply:
Typo. Thanks!